test/test_malicious_html.rb

```diff
@@ -0,0 +1,81 @@
+# encoding: utf-8
+require_relative 'common'
+
+# Miscellaneous attempts to sneak maliciously crafted HTML past Sanitize. Many
+# of these are courtesy of (or inspired by) the OWASP XSS Filter Evasion Cheat
+# Sheet.
+#
+# https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet
```

Comment:
This comment was backported from the 4.x branch, but it's not accurate in this branch since the tests it refers to weren't backported. It should probably be removed to prevent confusion. 

Alternatively, the new tests for the libxml2 fix could simply be added to the "bugs" section at the end of `test_sanitize.rb`, which may be a more suitable place for them in this branch.